File: model-client/src/main/java/org/modelix/model/persistent/CPHamtLeaf.java
package org.modelix.model.persistent;




public class CPHamtLeaf extends CPHamtNode {

  public final long key;

  /**
   * SHA to CPElement
   */
  public final String value;

  public CPHamtLeaf(long key, String value) {
    this.key = key;
    this.value = value;
  }

  public long getKey() {
    return key;
  }

  public String getValue() {
    return value;
  }

  @Override
  public String serialize() {
    return "L/" + SerializationUtil.longToHex(key) + "/" + value;
  }
}

File: model-client/src/main/kotlin/org/modelix/model/persistent/CPHamtLeaf.kt
package org.modelix.model.persistent

import org.modelix.model.persistent.SerializationUtil.longToHex

class CPHamtLeaf(
    val key: Long,
    /**
     * SHA to CPElement
     */
    val value: String
) : CPHamtNode() {

    override fun serialize(): String {
        return "L/" + longToHex(key) + "/" + value
    }
}
